internal/faustjs.org/docs/gutenberg/tutorial/create-a-custom-block.mdx

```diff
@@ -0,0 +1,314 @@
+
+For this step, we will make a simple block named MyCoolBlock, which has a RichText attribute and a radial button attribute from the @wordpress/components library.
+
+## 4. Create the initial block
+
+Inside the codebase of your application, create the basic block structure that simply prints the block's attributes.
+The steps required to achieve that are the following:
+
+### 4.1 Save it inside the `wp-blocks` folder
+
+```jsx title="wp-blocks/DnDBlock.js"
+export default function MyCoolBlock(props) {
+  console.log(props.attributes);
+  return <div>MyCoolBlock</div>
+}
+```
+
+This component prints the block attributes via the props passed as a parameter.
+
+### 4.2 Export the block in the `index.js`
+
+```jsx title="wp-blocks/index.js"
+import MyCoolBlock from './MyCoolBlock';
+export default {
+  MyCoolBlock
+}
+```
+
+By exporting the block in the `index.js` we make it available in the `WordPressBlocksProvider`.
+
+### 4.3 Create the block fragment
+
+You want to create a fragment that describes the request block fields and attributes. Ideally you want to include all the relevant fields for the implementation part:
+
+```jsx title="wp-blocks/MyCoolBlock.js" {1}
+import {gql} from '@apollo/client';
+...
+MyCoolBlock.fragments = {
+  key: `MyCoolBlockBlockFragment`,
+  entry: gql`
+    fragment MyCoolBlockBlockFragment on MyCoolBlock {
+      attributes {
+        class
+        content
+        message
+      }
+    }
+  `,
+};
+```
+We chose to attach the fragment as a property of the function component itself. This follows best practice for creating [colocated fragments](https://www.apollographql.com/docs/react/data/fragments/#creating-colocated-fragments).
+
+### 4.4 Include the block fragment into the page query string
+
+You want to include the fragment in the page query string and reference the block fragment key so that when the page resolves, the payload includes the data from that block.
+
+In our example we use it in both the Posts and Pages templates when using the [Faust Template Hierarchy](/docs/templates):
+
+```jsx {1,7,18} title="wp-templates/single.js"
+import blocks from '../wp-blocks';
+export default function Component(props) {
+...
+}
+Component.query = gql`
+...
+${blocks.MyCoolBlock.fragments.entry}
+query GetPost(
+...
+	post(id: $databaseId, idType: DATABASE_ID, asPreview: $asPreview) {
+		...
+		contentBlocks {
+			name
+			__typename
+			renderedHtml
+			id: nodeId
+			parentId
+			${blocks.MyCoolBlock.fragments.key}
+      	}
+	}
+...
+}
+```
+
+If you navigate to the page that contains this block, you should be able to inspect the properties in the console:
+
+<img
+    src="/docs/img/core-code-block-console.png"
+    alt="Code Code preview of the block attributes in the console"
```

Comment:
I think here people are going to get confused since they are seeing the wrong attributes for a different block.They should be seeing a simple `content` attribute.

 Maybe you can create a new screenshot.

Reply:
Will do! I'll just keep this here for now for my placeholder. Thanks Theo!